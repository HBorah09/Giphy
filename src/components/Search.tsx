import { useState, FC } from 'react';
import { Container, Row, Col ,Form, Button } from 'react-bootstrap';

interface Props {
    handleSearch: (str: string) => void
}

const Search: FC<Props> = ({ handleSearch }) => {
    const [search, setSearch] = useState("");
    const [error, setError] = useState(false);

    const handleSubmit = (e: React.MouseEvent<HTMLButtonElement>) => {
        e.preventDefault();
        if(search) {
            handleSearch(search);
        } else setError(true);
    }

    const handleChange = (value: string) => {
        setSearch(value);
        setError(false);
    }

    return  (
        <Container>
            <Form>
                <Row>
                    <Col xs={8}>
                        <Form.Group>
                            <Form.Control placeholder='Search a gif' onChange={e => handleChange(e.target.value)} />
                        </Form.Group>
                        {error && <span>Please enter search value</span>}
                    </Col>
                    <Col xs={4}>
                        <Button variant="primary" type="submit" onClick={e => handleSubmit(e)}>Submit</Button>
                    </Col>
                </Row>
            </Form>
        </Container>
    )
}

export default Search;