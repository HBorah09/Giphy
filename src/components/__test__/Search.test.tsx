import { shallow } from 'enzyme'
import Search from '../Search';

describe('<Search />', () => {

    const props = {
        handleSearch: () => jest.fn()
    }

    it('renders without crashing', () => {
        const wrapper = shallow(<Search { ...props } />);
        expect(wrapper).toMatchSnapshot();
    });
})