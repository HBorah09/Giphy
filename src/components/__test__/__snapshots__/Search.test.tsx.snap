// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Search /> renders without crashing 1`] = `
<Container
  fluid={false}
>
  <Form>
    <Row>
      <Col
        xs={8}
      >
        <FormGroup>
          <FormControl
            onChange={[Function]}
            placeholder="Search a gif"
          />
        </FormGroup>
      </Col>
      <Col
        xs={4}
      >
        <Button
          active={false}
          disabled={false}
          onClick={[Function]}
          type="submit"
          variant="primary"
        >
          Submit
        </Button>
      </Col>
    </Row>
  </Form>
</Container>
`;
